### PR TITLE
Fix operator crash on invalid CNP

### PR DESCRIPTION
**Pull request by eloycoto**
1) Fix crash in operator, handle errors in a common package. 
2) Fix typo on operator dockerfile.

---
This change is [Reviewable](https://reviewable.io/reviews/cilium/cilium/6913)

### CODE REVIEW

**Comment by eloycoto** [comment]
test-me-please

File: cilium-operator.Dockerfile [hunk 1]
```diff
@@ -11,4 +11,4 @@ FROM scratch
 LABEL maintainer="maintainer@cilium.io"
 COPY --from=builder /go/src/github.com/cilium/cilium/operator/cilium-operator /usr/bin/cilium-operator
 WORKDIR /
-CMD ["/usr/bin/cilium-etcd-operator"]
+CMD ["/usr/bin/ciliumt-operator"]
```
Comment (aanm) [inline cilium-operator.Dockerfile]: I think this might have make it worse :)

**Review by aanm — CHANGES_REQUESTED** [review]

File: cilium-operator.Dockerfile (hunk 1 above)
Comment (eloycoto) [reply cilium-operator.Dockerfile]: :man_facepalming:

**Review by aanm — APPROVED** [review]

**Comment by eloycoto** [comment]
test-me-please

**Comment by eloycoto** [comment]
test-me-please

**Comment by tgraf** [comment]
test-me-please

**Review by tgraf — APPROVED** [review]

**Review by ianvernon — APPROVED** [review]

**Pull request merged**